### PR TITLE
`StringIndexOutOfBoundsException` in `AbstractItem.getUrl`

### PR DESCRIPTION
**Pull request by jglick**
Noticed numerous stack traces in a log file of the form

```
WARNING	h.ExpressionFactory2$JexlExpression#evaluate: Caught exception evaluating: item.task.url in /controllername/. Reason: java.lang.reflect.InvocationTargetException
java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.base/java.lang.String.substring(String.java:1841)
	at hudson.model.AbstractItem.getUrl(AbstractItem.java:559)
	at hudson.model.AbstractItem.getUrl(AbstractItem.java:575)
	at hudson.model.Run.getUrl(Run.java:1055)
	at org.jenkinsci.plugins.workflow.support.steps.ExecutorStepExecution$PlaceholderTask.getUrl(ExecutorStepExecution.java:655)
```

Does not seem like it should be possible given `Ancestor.getUrl` Javadoc, but there it is. This whole block of logic from ea95434938b2111d8768528823990a78bdc58d3e is optional—we should not throw an exception if it does not work out.

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8481"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
